### PR TITLE
Get rid off some legacy vim syntax, added deprecation warnings and test.

### PR DESCRIPTION
**Pull request by ubaldot**
Some vim legacy leftover have been adjusted. 

Q1: Are legacy v:true/v:false equivalent to Vim9 true/false?
Q2: Would that be possible to rewrite the termdebug tests in Vim9?

### CODE REVIEW

File: runtime/doc/terminal.txt [hunk 1]
```diff
@@ -1530,9 +1530,9 @@ in a buffer with 'buftype' set to "prompt".  This works slightly differently:
 
 						*termdebug_use_prompt*
 Prompt mode can be used even when the |+terminal| feature is present with: >
-	let g:termdebug_config['use_prompt'] = 1
+	let g:termdebug_config['use_prompt'] = true
```
Comment (chrisbra) [inline runtime/doc/terminal.txt]: This uses legacy vim script syntax, so should be `v:true`

File: runtime/doc/terminal.txt [hunk 2]
```diff
@@ -1530,9 +1530,9 @@ in a buffer with 'buftype' set to "prompt".  This works slightly differently:
...
 If there is no g:termdebug_config you can use: >
-	let g:termdebug_use_prompt = 1
+	let g:termdebug_use_prompt = true
```
Comment (chrisbra) [inline runtime/doc/terminal.txt]: same here

File: runtime/doc/terminal.txt [hunk 3]
```diff
@@ -1544,26 +1544,26 @@ are reset to their original values once the termdebug session concludes.
 					*termdebug_map_K* *termdebug-mappings*
 The K key is normally mapped to |:Evaluate| unless a buffer local (|:map-local|)
 mapping to K already exists.  If you do not want this use: >
-	let g:termdebug_config['map_K'] = 0
+	let g:termdebug_config['map_K'] = false
```
Comment (chrisbra) [inline runtime/doc/terminal.txt]: and here

File: runtime/pack/dist/opt/termdebug/plugin/termdebug.vim [hunk 4]
```diff
@@ -63,7 +63,7 @@ command -nargs=+ -complete=file -bang TermdebugCommand StartDebugCommand(<bang>0
 
 enum Way
   Prompt,
-  Terminal
+    Terminal
```
Comment (chrisbra) [inline runtime/pack/dist/opt/termdebug/plugin/termdebug.vim]: white space error

File: runtime/pack/dist/opt/termdebug/plugin/termdebug.vim [hunk 5]
```diff
@@ -655,16 +667,16 @@ def StartDebug_prompt(dict: dict<any>)
...
+    out_cb: GdbOutCallback
   })
   if job_status(gdbjob) != "run"
     Echoerr('Failed to start gdb')
     exe $'bwipe! {promptbufnr}'
     return
   endif
   exe $'au BufUnload <buffer={promptbufnr}> ++once ' ..
-       'call job_stop(gdbjob, ''kill'')'
+    'job_stop(gdbjob, ''kill'')'
```
Comment (chrisbra) [inline runtime/pack/dist/opt/termdebug/plugin/termdebug.vim]: I am not sure this is correct. Are you sure the exec'ed string is evaluated in vim9 script semantics?

**Comment by chrisbra** [comment]
Thanks. In general, it would be beneficial to have separate commits for 
- cleaning up white space
- adding new functions (e.g. for the deprecation warning)
- changing arguments to true/false

That makes reviewing a bit easier.

**Comment by ubaldot** [comment]
You are right, sorry!

File: runtime/pack/dist/opt/termdebug/plugin/termdebug.vim (hunk 5 above)
Comment (ubaldot) [reply runtime/pack/dist/opt/termdebug/plugin/termdebug.vim]: Not really... I replaced it back.

**Comment by chrisbra** [comment]
thanks!

**Pull request closed without merging**